### PR TITLE
fetch user groups while building user object

### PR DESCRIPTION
**Pull request by Happycoil**
Signed-off-by: Tor Magnus Rakvåg <tm@intility.no>

Fixes https://github.com/chef/inspec/issues/1400

### CODE REVIEW

**Review by adamleff — APPROVED** [review]
@Happycoil Thank you for your contribution! I'm not a big PowerShell user but this looks OK to me.

Also, thanks for cleaning up some whitespace consistency issues. In the future, it's incredibly helpful to do all the "code cleanup" work in a separate commit on the same PR so it's easier to separate the "style" changes from the material "code" changes. We can take care of squashing those commits if deemed necessary before merge. But stuff like that is nicer as separate commits.

Thanks again for submitting this PR. Looking to @chris-rock or @arlimus for additional feedback as they have more experience with this resource and PowerShell than I do.

File: test/integration/default/user_spec.rb [hunk 1]
```diff
@@ -91,13 +91,15 @@
     it { should exist }
     # should return the SID of the user
     its('uid') { should_not eq nil}
+    its('groups') { should include 'Administrators' }
```
Comment (chris-rock) [inline test/integration/default/user_spec.rb]: this should be
```
its('groups') { should include userinfo[:username] }

```

File: test/integration/default/user_spec.rb [hunk 2]
```diff
@@ -91,13 +91,15 @@
...
   end
 
   # also support simple username for local users without domain
   describe user('Administrator') do
     it { should exist }
     # should return the SID of the user
     its('uid') { should_not eq nil}
+    its('groups') { should include 'Administrators' }
```
Comment (chris-rock) [inline test/integration/default/user_spec.rb]: Same as above

File: test/unit/resources/user_test.rb [hunk 3]
```diff
@@ -103,7 +103,7 @@
     _(resource.uid).wont_be_nil
     _(resource.exists?).must_equal true
     _(resource.group).must_equal nil
-    _(resource.groups).must_equal nil
+    _(resource.groups).must_equal ['Administrators', 'Users']
```
Comment (chris-rock) [inline test/unit/resources/user_test.rb:106]: We should add a test for `Guest` user to see that we receive `Users` only

**Review by chris-rock — COMMENTED** [review]
This is a great addition @Happycoil Thank you! I left some minor improvements.

File: test/integration/default/user_spec.rb (hunk 1 above)
Comment (Happycoil) [reply test/integration/default/user_spec.rb]: Fixed.

File: test/integration/default/user_spec.rb (hunk 2 above)
Comment (Happycoil) [reply test/integration/default/user_spec.rb]: Fixed.

File: test/unit/resources/user_test.rb (hunk 3 above)
Comment (Happycoil) [reply test/unit/resources/user_test.rb:106]: Fixed.

**Review by chris-rock — APPROVED** [review]
Thank you @Happycoil

**Pull request merged**